### ISSUE TITLE
ReportFrameSavingPath doesn't work on F4i detector

### ISSUE THREAD

**Issue by xtalfire** [eisfabian/PACEtomo#30]
Hi Fabian,

Just a report. I recently updated PACEtomo to the latest version1.8 on our Krios with F4i detector. The script looks for frame path by serialEM command "ReportFrameSavingPath" , but it doesn't work for F4i in our system. Every time the PACEtomo script starts SerialEM produce a box saying "ReportFrameSavingPath only works with Gatan or generic frame saving cameras" that needs to be clicked. The script runs fine after the box is clicked, but it needs a click for every item to be acquired, which means every PACEtomo group needs a click which is a nuisance. 

My current workaround is to disable the line "framePath = sem.ReportFrameSavingPath()" , instead I set framePath to be a variable which refers to the frame path. Then the script runs fine. 

Best,
Tat

**Comment by eisfabian** [comment]
Dear Tat,

Thank you for reporting the issue! In the version you were using, was the ReportFrameSavingCommand already in a try block?

```
try:
    framePath = sem.ReportFrameSavingPath()
except:
    log("WARNING: Camera frame path could not be obtained for your camera.")
    framePath = "NONE"
```

I was hoping that this would haven taken care of the issue you describe, but I might have to add some more checks.

Thanks and best wishes,
Fabian

**Comment by xtalfire** [comment]
Hi Fabian,

Thanks for the message. I simply used the latest version you posted, so the try clause is there as you indicated. Nonetheless the serialEM command ReportFrameSavingPath triggers a click box always somehow for our camera. I tested the command standalone on SerialEM and it triggered the same click box. 

Tat

**Comment by eisfabian** [comment]
Hi Tat,

Could you try replacing the try block with this one?

```
try:
    sem.NoMessageBoxOnError()
    framePath = sem.ReportFrameSavingPath()
except:
    log("WARNING: Camera frame path could not be obtained for your camera.")
    framePath = "NONE"
finally:
    sem.NoMessageBoxOnError(0)
```

That should suppress the box and let the script continue.
I am still not sure why this command does not work with the Falcon4i. Did you set a frame path?

Best,
Fabian

**Comment by xtalfire** [comment]
Hi Fabian,

Thanks for the help. I just tried to the new try block. It works in the sense that now the error goes to the log instead of making a click box. The error message ( "ReportFrameSavingPath only works with Gatan or generic frame saving cameras" ) is still the same but it only appears in the log without the need of user clicking it. 
We set up the frame path in the camera dialogue as usual, and the eer output are exactly where it should be. Somehow this SerialEM command just doesn't work with F4. I asked Chen Xu about it, he replied that K3 frame path can be flexible, whereas F4i frame path has to be in the designated location. 

Tat
